vendor/github.com/OpenBazaar/multiwallet/client/blockbook/client.go

```diff
@@ -503,18 +504,28 @@ func (i *BlockBookClient) TransactionNotify() <-chan model.Transaction {
 	return i.txNotifyChan
 }
 
-func (i *BlockBookClient) ListenAddress(addr btcutil.Address) {
+func (i *BlockBookClient) ListenAddresses(addrs ...btcutil.Address) {
 	i.listenLock.Lock()
 	defer i.listenLock.Unlock()
 	var args []interface{}
 	args = append(args, "bitcoind/addresstxid")
-	args = append(args, []string{maybeConvertCashAddress(addr)})
+
+	var convertedAddrs []string
+	for _, addr := range addrs {
+		convertedAddrs = append(convertedAddrs, maybeConvertCashAddress(addr))
+	}
+
+	i.listenAddrs = convertedAddrs
+
+	args = append(args, convertedAddrs)
 	i.socketMutex.RLock()
 	defer i.socketMutex.RUnlock()
 	if i.SocketClient != nil {
 		i.SocketClient.Emit("subscribe", args)
```

Comment:
```
i.listenAddrs = convertedAddrs
args = append(args, convertedAddrs)
```
needs to be inside the `i.SocketClient != nil` case so it's not added to `listenAddrs` at the same time as `listenQueue`.

Reply:
Fixed